src/docs/platform/guides/custom-binary-backends.md

```diff
@@ -0,0 +1,171 @@
+# Using your own storage
+
+This guide walks you through setting up custom binary backends for Peridio, allowing you to use your own storage infrastructure instead of Peridio's default storage.
+
+## Overview
+
+Custom binary backends let you store your binaries in your own cloud storage while still using Peridio's device management capabilities. This is useful for:
+
+- **Compliance requirements** - Keep your binaries in specific regions or providers
+- **Cost optimization** - Use your existing storage contracts
+- **Performance** - Store binaries closer to your devices
+- **Security** - Maintain full control over your binary storage
+
+## Supported Backends
+
+Currently, Peridio supports:
+- **Amazon S3** - Direct storage in your S3 buckets
+- **Amazon CloudFront** - CDN distribution with signed URLs for enhanced security and performance
+
+## Setting Up Amazon S3 Backend
+
+### Prerequisites
+
+Before you begin, make sure you have:
+- An AWS account with administrative access
+- A Peridio organization set up
+- Basic familiarity with AWS IAM and S3
+
+### Step 1: Create Your S3 Bucket
+
+1. Log into your AWS Console
+2. Navigate to S3 and create a new bucket
+3. Choose a unique bucket name (you'll need this later)
+4. Select the AWS region where you want to store your binaries
+5. Keep the default settings for now - we'll configure permissions next
+
+### Step 2: Create an IAM Role for Peridio
+
+Peridio needs permission to upload, download, and manage binaries in your bucket. Let's create a dedicated IAM role:
+
+1. Go to AWS IAM console
+2. Click "Roles" → "Create role"
+3. Choose "AWS account" as the trusted entity type
+4. For now, enter a temporary AWS account ID (we'll update this later)
+5. Name your role something descriptive
+
+### Step 3: Attach the Bucket Policy
+
+Your IAM role needs specific permissions to work with your S3 bucket. Attach this policy to your role:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:GetObject",
+                "s3:PutObject",
+                "s3:DeleteObject",
+                "s3:ListBucket"
+            ],
+            "Resource": [
+                "arn:aws:s3:::YOUR_BUCKET_NAME",
+                "arn:aws:s3:::YOUR_BUCKET_NAME/*"
+            ]
+        }
+    ]
+}
+```
+
+**Important**: Replace `YOUR_BUCKET_NAME` with your actual bucket name.
+
+### Step 4: Register with Peridio
+
+Contact Peridio support with the following information:
+
+**Required Information:**
+- **Bucket name** - The name of your S3 bucket
+- **AWS region** - The region where your bucket is located
+- **Role ARN** - The Amazon Resource Name of the IAM role you created
+- **Role name** - A short name for Peridio's internal use (max 10 characters - this doesn't need to match your IAM role name)
```

Comment:
If this is for "Peridio's internal use", and has no link to the actual role name, why let the customer dictate it? Is it just so we don't have to come up with our own 10-char limited naming scheme?

Reply:
- There is a future case where a user can have multiple backends. So this could helpful for identifying quickly.
- This name comes back to you when are updating the permissions so you could correlate data.

Maybe a better description?

Reply:
Updated